python/ray/ml/examples/tensorflow/tensorflow_linear_dataset_example.py

```diff
@@ -0,0 +1,149 @@
+import argparse
+import numpy as np
+
+
+import tensorflow as tf
+from tensorflow.keras.callbacks import Callback
+
+import ray
+import ray.train as train
+from ray.data import Dataset
+from ray.data.dataset_pipeline import DatasetPipeline
+from ray.train.tensorflow import prepare_dataset_shard
+from ray.ml.train.integrations.tensorflow import TensorflowTrainer
+from ray.ml.predictors.integrations.tensorflow import TensorflowPredictor
+from ray.ml.constants import MODEL_KEY, TRAIN_DATASET_KEY
+from ray.ml.result import Result
+
+
+class TrainCheckpointReportCallback(Callback):
+    def on_epoch_end(self, epoch, logs=None):
+        train.save_checkpoint(**{MODEL_KEY: self.model.get_weights()})
+        train.report(**logs)
+
+
+def get_dataset_pipeline(a=5, b=10, size=1000) -> DatasetPipeline:
+    def get_dataset(a, b, size) -> Dataset:
+        items = [i / size for i in range(size)]
+        dataset = ray.data.from_items([{"x": x, "y": a * x + b} for x in items])
+        return dataset
+
+    dataset = get_dataset(a, b, size)
+    train_dataset_pipeline = dataset.repeat().random_shuffle_each_window()
```

Comment:
Trainer is not supposed to support DatasetPipelines for now. Can we change this to a regular Dataset (remove the repeat and random shuffle)?